File: ADS-2-Semestre/EngenhariaDeSoftware/InterfacePatoCorredor/Pato_Corrredor.java
package engsoft;

public class Pato_Corrredor extends Pato implements  Padrao_Grasnar{
   
    public Pato_Corrredor()	
	{
		setComportamentoCorrida(new Correr());		
	}
	
	public String mostrar() {		
		return "Eu sou o Pato Corredor.";
	}

	public String grasnar() {
		return "Que-Que. Grrrrrrrrr.";
	}
}

File: ADS-2-Semestre/EngenhariaDeSoftware/InterfacePatoC#/Pato/PatoCorrredor.cs
using Pato;
using System;
using System.Collections.Generic;
using System.Text;

namespace JogoPato
{
	public class PatoCorrredor: Pato,IPadraoGrasnar
    {
		public PatoCorrredor()
		{
			setComportamentoCorrida(new Correr());
		}

		public String grasnar()
		{
			return "Que-Que. Grrrrrrrrr.";
		}

		public override string mostrar()
		{
			return "Eu sou o Pato Corredor.";
		}
	}
}
